app/src/main/java/org/dhis2/Bindings/ViewExtensions.kt

```diff
@@ -69,3 +74,43 @@ fun View.clipWithAllRoundedCorners(curvedRadio: Int = 16.dp) {
         clipToOutline = true
     }
 }
+
+fun Spinner.overrideHeight(desiredHeight: Int) {
+    try {
+        val popup = Spinner::class.java.getDeclaredField("mPopup")
+        popup.isAccessible = true
+
+        // Get private mPopup member variable and try cast to ListPopupWindow
+        val popupWindow = popup[this] as ListPopupWindow
+
+        // Set popupWindow height
+        popupWindow.height = desiredHeight
+    } catch (e: Exception) {
+        // silently fail...
+    }
+}
+
+fun Spinner.doOnItemSelected(onItemSelected: (selectedIndex: Int) -> Unit) {
+    onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+        override fun onItemSelected(
+            adapterView: AdapterView<*>?,
+            view: View?,
+            selectedIndex: Int,
+            id: Long
+        ) {
+            onItemSelected(selectedIndex)
+        }
+
+        override fun onNothingSelected(p0: AdapterView<*>?) {
+            // Don't do anything
+        }
+    }
+}
+
+fun FloatingActionButton.display(shouldBeDisplayed: Boolean) {
```

Comment:
not sure, but could this be replaced by the visible property of the floating action button?